src/Vibrato/Vibrato.cpp

```diff
@@ -0,0 +1,194 @@
+#include "Vibrato.h"
+
+CVibrato::CVibrato(float fDelayInSec, float fWidthInSec, float fSampleRateInHz, float fFrequencyInHz, int numChannels): m_isInitialised(false)
+{
+    init(fDelayInSec, fWidthInSec, fSampleRateInHz, fFrequencyInHz, numChannels);
+}
+
+CVibrato::~CVibrato()
+{
+    this->reset();
+}
+
+Error_t CVibrato::create (CVibrato*& pCInstance, float fDelayInSec, float fWidthInSec, float fSampleRateInHz, float fFrequencyInHz, int numChannels )
+{
+    pCInstance = new CVibrato(fDelayInSec, fWidthInSec, fSampleRateInHz, fFrequencyInHz, numChannels );
+    return Error_t::kNoError;
+}
+
+Error_t CVibrato::destroy (CVibrato*& pCInstance)
+{
+    if (!pCInstance)
+        return Error_t::kUnknownError;
+    delete pCInstance;
+    pCInstance = nullptr;
+    return Error_t::kNoError;
+}
+
+Error_t CVibrato::init(float fDelayInSec, float fWidthInSec, float fSampleRateInHz, float fFrequencyInHz, int numChannels )
+{
+        m_fSampleRateInHz = fSampleRateInHz;
+        m_iNumChannels = numChannels;
+
+        setParam(kDelay, fDelayInSec);
+        setParam(kWidth, fWidthInSec);
+
+//        this->reset();
+        //Initialise and set memory for the vibrato ring buffer
+        if(ringBuffer != nullptr)
+        {
+            for(int i=0;i<m_iNumChannels;i++)
+            {
+                delete ringBuffer[i];
+            }
+            delete[] ringBuffer;
+            ringBuffer = nullptr;
+        }
+        int bufferSize = 2 + m_iDelayInSamples + m_iWidthInSamples * 2;
+        ringBuffer = new CRingBuffer<float> *[m_iNumChannels];
+        for (int i = 0; i < m_iNumChannels; i++)
+        {
+            ringBuffer[i] = new CRingBuffer<float>(bufferSize);
+        }
+        if (lfo != nullptr)
```

Comment:
The next few lines could've been avoided by just using this->reset() just as Jocelyn mentioned!